src/cbmc/c_test_input_generator.cpp

```diff
@@ -0,0 +1,176 @@
+/*******************************************************************\
+
+Module: Test Input Generator for C
+
+Author: Daniel Kroening, Peter Schrammel
+
+\*******************************************************************/
+
+/// \file
+/// Test Input Generator for C
+
+#include "c_test_input_generator.h"
+
+#include <goto-checker/goto_trace_storage.h>
+
+#include <langapi/language_util.h>
+
+#include <util/json.h>
+#include <util/json_irep.h>
+#include <util/options.h>
+#include <util/xml.h>
+#include <util/xml_irep.h>
+
+#include <goto-programs/json_expr.h>
+#include <goto-programs/json_goto_trace.h>
+#include <goto-programs/xml_expr.h>
+#include <goto-programs/xml_goto_trace.h>
+
+c_test_input_generatort::c_test_input_generatort(
+  ui_message_handlert &ui_message_handler,
+  const optionst &options)
+  : ui_message_handler(ui_message_handler), options(options)
+{
+}
+
+void test_inputst::output_plain_text(
+  std::ostream &out,
+  const namespacet &ns,
+  const goto_tracet &goto_trace,
+  bool print_trace) const
+{
+  bool first = true;
+  for(const auto &step : goto_trace.steps)
+  {
+    if(step.is_input())
+    {
+      if(first)
+        first = false;
+      else
+        out << ", ";
+
+      out << id2string(step.io_id) << "=";
+
+      if(step.io_args.size() == 1)
+        out << from_expr(ns, "", step.io_args.front());
+    }
+  }
```

Comment:
Nowadays I'd actually use
```
const auto range = make_range(goto_trace.steps).filter(
  [](const goto_tracet::stept &step) { return step.is_input(); }).map(
    [](const goto_tracet::stept &step) {
      return id2string(step.io_id) + '=' + from_expr(ns, step.function_id, step.io_args.front()); });
  join_strings(out, range.begin(), range.end(), ", ");
```
but maybe that's actually less readable...

Either way, there are two problems here: 1) a weird/incomplete output in case `step.io_args.size() != 1` and 2) the use of `""` in `from_expr`, which should use `step.function_id` instead.